### PR TITLE
Publish daml-lf encoder, parser and error-test-lib [KVL-1424]

### PR DESCRIPTION
**Pull request by hubert-da**
I double-checked and we need all these 3 libraries to re-enable tests in kvutils.

### CODE REVIEW

File: daml-lf/encoder/BUILD.bazel [hunk 1]
```diff
@@ -25,6 +25,7 @@ da_scala_library(
         "@maven//:org_scalaz_scalaz_core",
     ],
     scalacopts = lf_scalacopts,
+    tags = ["maven_coordinates=com.daml:daml-lf-encoder:__VERSION__"],
```
Comment (hubert-da) [inline daml-lf/encoder/BUILD.bazel:28]: @remyhaemmerle-da are you fine with publishing these libs?

File: ledger/error/BUILD.bazel [hunk 2]
```diff
@@ -41,7 +41,6 @@ da_scala_library(
     name = "error-test-utils",
     srcs = glob(["src/test/utils/**/*.scala"]),
     scala_deps = [],
-    tags = ["maven_coordinates=com.daml:error-test-package:__VERSION__"],
```
Comment (hubert-da) [inline ledger/error/BUILD.bazel:44]: I don't think we used it.

File: daml-lf/encoder/BUILD.bazel (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/encoder/BUILD.bazel:28]: I don't want this to be published. 
I do not want anybody outside of DA to use it. 

I see two solution. 
- You generate your `dar` file in the main repo, for instance in the //daml-lf/kv-support 
- You rewrite you test in daml.

Comment (remyhaemmerle-da) [reply daml-lf/encoder/BUILD.bazel:28]: I am very sorry for the inconvenient.

**Review by remyhaemmerle-da — CHANGES_REQUESTED** [review]
We do not want the encoder to be publicly available. The main reasons are :

- We want to discourage as much as possible people to create **production** dars without passing through `damlc`. 
- We do not want to offer any grantee about the encoder, to anybody outside DA.

**Review by remyhaemmerle-da — APPROVED** [review]
On reflection, I will let you publish the libraries. 
I add two `README.md` to make explicit no guarantees are provided for those two libraries.

**Pull request merged**